ancestrymanager/ancestrymanager.go

```diff
@@ -313,3 +312,25 @@ type NoOpAncestryManager struct{}
 func (*NoOpAncestryManager) Ancestors(config *resources.Config, tfData resources.TerraformResourceData, cai *resources.Asset) ([]string, string, error) {
 	return nil, "", nil
 }
+
+func (m *manager) handleUnknownProject(config *resources.Config, tfData resources.TerraformResourceData, cai *resources.Asset) ([]string, error) {
```

Comment:
It feels like this duplicates some of the logic in fetchAncestors - I wonder if it would be possible to refactor fetchAncestors so that this function isn't required, now that partial ancestries are allowed? For example, maybe fetchAncestors could get the project, folder, and organization from the current resource, and then build the ancestry in the same way for all resources, like:

1. get org, folder, and project from resource
2. If project is set, get project ancestry
   - If folder / org are set, and match the fetched ancestry, return it.
   - Returns 403 errors from the API
3. If folder is set, get folder ancestry
   - Returns 403 errors from the API
4. Combine project + folder / org ancestries to get full resource ancestry
5. If no ancestry could be built, return "organizations/unknown"

What do you think - would something like that work / be simpler? Would we potentially run into issues with resource-specific behavior?

Reply:
I changed the code to basically follow this plan, but it mostly apples to project resource only, and not expanding to other resource types.

Reply:
👍  sounds good & looks good